core/circuitbreaker/circuit_breaker.go

```diff
@@ -135,13 +136,25 @@ func (b *circuitBreakerBase) fromClosedToOpen(snapshot interface{}) bool {
 	return false
 }
 
+const halfOpenCircuitBreakerKey = "__circuit_breaker_state_revise__"
+
 // fromOpenToHalfOpen updates circuit breaker state machine from open to half-open.
 // Return true only if current goroutine successfully accomplished the transformation.
-func (b *circuitBreakerBase) fromOpenToHalfOpen() bool {
+func (b *circuitBreakerBase) fromOpenToHalfOpen(ctx *base.EntryContext) bool {
 	if b.state.casState(Open, HalfOpen) {
 		for _, listener := range stateChangeListeners {
 			listener.OnTransformToHalfOpen(Open, b.rule)
 		}
+
+		value, exist := ctx.Data[halfOpenCircuitBreakerKey]
```

Comment:
Using ctx.Data maybe not very elegant.
Maybe it's the other form that's stored in the context/entry.